azkaban-hadoop-security-plugin/src/main/java/azkaban/security/AbstractHadoopSecurityManager.java

```diff
@@ -0,0 +1,698 @@
+   * containerized execution. The KeyStore object acquired is placed in KeyStoreManager for future
+   * use.
+   *
+   * @param props Azkaban Props containing CredentialProvider info.
+   * @return KeyStore object.
+   */
+  @Override
+  public KeyStore getKeyStore(final Props props) {
+    logger.info("Prefetching KeyStore for the flow");
+    final Credentials cred = new Credentials();
+    final CredentialProviderWithKeyStore customCredential = (CredentialProviderWithKeyStore)
+        getCustomCredentialProvider(props, cred, logger,
+            Constants.ConfigurationKeys.CUSTOM_CREDENTIAL_NAME);
+    final KeyStore keyStore = customCredential.getKeyStore();
+    KeyStoreManager.getInstance().setKeyStore(keyStore);
+    return keyStore;
+  }
+
+  /**
+   * This method is used to verify whether Hadoop security is enabled or not.
+   * @return
+   */
+  @Override
+  public boolean isHadoopSecurityEnabled() {
+    return this.securityEnabled;
+  }
+
+  /*
+   * Gets hadoop tokens for a user to run mapred/hive jobs on a secured cluster
+   */
+  @Override
+  public void prefetchToken(final File tokenFile, final Props props, final Logger logger)
+      throws HadoopSecurityManagerException {
+    final String userToProxy = props.getString(JobProperties.USER_TO_PROXY);
+
+    doPrefetch(tokenFile, props, logger, userToProxy);
+  }
+
+  /**
+   * This method is used to prefetch all required tokens for a job.
+   * @param tokenFile
+   * @param props
+   * @param logger
+   * @param userToProxy
+   * @throws HadoopSecurityManagerException
+   */
+  protected void doPrefetch(final File tokenFile, final Props props, final Logger logger,
+      final String userToProxy) throws HadoopSecurityManagerException {
+    // Create suffix to be added to kerberos principal
+    final String suffix =
+        (null != props.getString(HadoopSecurityManager.DOMAIN_NAME, null)) ?
+            "/" + kerberosSuffix(props) : "";
+
+    final String userToProxyFQN = userToProxy + suffix;
+    logger.info("Getting hadoop tokens based on props for " + userToProxyFQN);
+
+    final Credentials cred = new Credentials();
+    fetchMetaStoreToken(userToProxyFQN, userToProxy, props, logger, cred);
+
+    try {
+      fetchJHSToken(userToProxyFQN, userToProxy, props, logger, cred);
+
+      getProxiedUser(userToProxyFQN).doAs(new PrivilegedExceptionAction<Void>() {
+        @Override
+        public Void run() throws Exception {
+          getToken(userToProxyFQN, userToProxy);
+          return null;
+        }
+
+        private void getToken(final String userToProxyFQN, final String userToProxy)
+            throws InterruptedException, IOException, HadoopSecurityManagerException {
+          logger.info("Here is the props for " + HadoopSecurityManager.OBTAIN_NAMENODE_TOKEN +
+              ": " + props.getBoolean(HadoopSecurityManager.OBTAIN_NAMENODE_TOKEN));
+
+          registerAllCustomCredentials(userToProxy, props, cred, logger);
```

Comment:
In Hadoop Security Manager 2.0, this code is called ifL ENABLE_OAUTH is false. Why the change?

Reply:
I have kept that check inside registerAllCustomCredentials. I have created registerAllCustomCredentials to take all the customCredentials related part out in separate method.